packages/neon/neon/lib/src/utils/result.dart

```diff
@@ -41,4 +41,21 @@ class Result<T> {
         loading: loading,
         cached: cached,
       );
+
+  Result<T> asLoading() => Result(
+        data,
+        error,
+        loading: true,
+        cached: cached,
+      );
+
+  bool get hasError => error != null;
+  bool get isLoading => loading;
```

Comment:
What about just renaming the variable? The additional getter doesn't add any functionality

Reply:
And then `cached` to `isCached` too.